scripts/local_k8s/deployment.py

```
@@ -2,39 +2,93 @@
+def _deploy_preview_tools(
+    config: PreviewConfig, *, skip_build: bool
+) -> tuple[str, ...]:
     """Return the required command-line tools for the requested deploy mode."""
 
+    match config.k8s_provider:
+        case "kind":
+            cluster_tool = "kind"
+        case "k3d":
+            cluster_tool = "k3d"
+        case unexpected:
+            raise LocalK8sError(f"Unsupported Kubernetes provider: {unexpected!r}")
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)